openvpn/config.sls

```diff
@@ -89,11 +89,17 @@ openvpn_config_{{ type }}_{{ name }}_passwd_file:
       - service: {{ service_id }}
 {% endif %}
 
-{% if config.tls_auth is defined and config.ta_content is defined %}
+{% if config.ta_content is defined %}
 # Deploy {{ type }} {{ name }} TLS key file
+  {% if config.tls_auth is defined and config.tls_crypt is not defined %}
```

Comment:
I'm thinking option 2, but we'll have to change `common_opts.jinja` accordingly.
Misconfiguration in Salt should lead to a crashing service (due to misconfiguration of the server) rather than a silent fail (neither `tls_auth` nor `tls_crypt` being used).
Are you ok with that?